hazelcast/src/main/java/com/hazelcast/spi/impl/operationexecutor/classic/ClassicOperationExecutor.java

```diff
@@ -148,27 +141,24 @@ private ResponseThread initResponseThread() {
             String threadName = threadGroup.getThreadPoolNamePrefix("partition-operation") + threadId;
             ScheduleQueue scheduleQueue = new DefaultScheduleQueue();
 
-            PartitionOperationThread operationThread = new PartitionOperationThread(
-                    threadName, threadId, scheduleQueue, logger,
+            PartitionOperationThread operationThread = new PartitionOperationThread(threadName, threadId, scheduleQueue, logger,
                     threadGroup, nodeExtension, partitionOperationRunners);
 
-            // we need to assign the thread to all OperationRunners this thread owns.
-            for (int partitionId = 0; partitionId < partitionOperationRunners.length; partitionId++) {
-                if (partitionId % threadCount == threadId) {
-                    OperationRunner operationRunner = partitionOperationRunners[partitionId];
-                    operationRunner.setCurrentThread(operationThread);
-                }
-            }
-
             threads[threadId] = operationThread;
             operationThread.start();
         }
 
+        // we need to assign the PartitionOperationThreads to all OperationRunners they own
+        for (int partitionId = 0; partitionId < partitionOperationRunners.length; partitionId++) {
+            int threadId = partitionId % threadCount;
+            partitionOperationRunners[partitionId].setCurrentThread(threads[threadId]);
```

Comment:
Can you make this line less complicated?

e.g.

```
Thread thread = threads[threadId]
PartitionOperationRunner runner = runners[partitionId];
runner.setThread(thread)
```
This is a lot easier to read.